lesson_07/homework_07.py

```diff
@@ -68,4 +122,56 @@ def find_substring(str1, str2):
 """  Оберіть будь-які 4 таски з попередніх домашніх робіт та
 перетворіть їх у 4 функції, що отримують значення та повертають результат.
 Обоязково документуйте функції та дайте зрозумілі імена змінним.
-"""
\ No newline at end of file
+"""
+
+
+def duplicate_list_checker(ch_list: list) -> bool:
+    """Checks the list for duplicates"""
+    ch_set = set(ch_list)
+    if len(ch_list) > len(ch_set):
+        return True
+    return False
+
+
+big_list = [3, 5, -2, -1, -3, 0, 1, 4, 5, 2]
+print("Duplicate checker:", duplicate_list_checker(big_list))
+
+
+def key_value_swapper(dictionary: dict) -> dict:
+    """Swaps key and value in the dictionary"""
+    new_dict = {}
+    for key, value in dictionary.items():
+        new_key = value if isinstance(value, str) else str(value)
```

Comment:
```suggestion
        new_key = value if isinstance(value, str) else str(value)
```
то всі валюї перетворить в стрінги

Reply:
не дуже зрозуміла що потрібно зробити. 
перетворити всі value в стрінг навіть якщо деякі value вже стрінг?

Reply:
ну ваш код саме це й робить, там по суті написано так:
`new_key = str(value)`
а повинно бути якщо значення вже є

Reply:
new_key = value if isinstance(value, str) else str(value)

а хіба ця строка не читається як "запиши в new_key значення value без змін, якщо value є строкою. В противному випадку, тобто якщо value не строка, то перетвори value на строку и запиши в new_key"?

ну тобто в моєму випадку str(value) відпрацює тільки один раз для ключа 'size' де значення int
base_dict = {'contry': 'Ukraine', 'continent': 'Europe', 'size': 123}
а value 'Ukraine' та 'Europe' він просто запише без перетворення в строку, бо це вже строка 

чи я не вірно розумію?